v0/destinations/google_adwords_offline_conversions/utils.js

```diff
@@ -0,0 +1,182 @@
+ * for OAuth destination
+ * get access_token from metadata.secret{ ... }
+ * @param {*} param0
+ * @returns
+ */
+const getAccessToken = ({ secret }) => {
+  if (!secret) {
+    throw new CustomError(
+      "[Google Ads Offline Conversions]:: OAuth - access token not found",
+      500
+    );
+  }
+  return secret.access_token;
+};
+
+/**
+ * This function helps to determine the type of error occured. We set the authErrorCategory
+ * as per the destination response that is received and take the decision whether
+ * to refresh the access_token or disable the destination.
+ * @param {*} code
+ * @param {*} response
+ * @returns
+ */
+const getAuthErrCategory = code => {
+  switch (code) {
+    case 401:
+      // UNAUTHORIZED
+      return REFRESH_TOKEN;
+    case 403:
+      // PERMISSION_DENIED
+      return DISABLE_DEST;
+    default:
+      return "";
+  }
+};
+
+/**
+ * get conversionActionId using the conversion name using searchStream endpoint
+ * @param {*} method
+ * @param {*} headers
+ * @param {*} params
+ * @returns
+ */
+const getConversionActionId = async (method, headers, params) => {
+  const conversionActionIdKey = sha256(
+    params.event + params.customerId
+  ).toString();
+  return conversionActionIdCache.get(conversionActionIdKey, async () => {
+    const data = {
+      query: `SELECT conversion_action.id FROM conversion_action WHERE conversion_action.name = '${params.event}'`
+    };
+    const requestBody = {
+      url: SEARCH_STREAM.replace(":customerId", params.customerId),
+      data,
+      headers,
+      method
+    };
+    const response = await httpSend(requestBody);
+    if (
+      !response.success &&
+      !isHttpStatusSuccess(response.response.response.status)
+    ) {
+      throw new ErrorBuilder()
+        .setStatus(response.response.response.status)
+        .setDestinationResponse(response.response.response.data)
+        .setMessage(
+          `[Google Ads Offline Conversions]:: ${response.response.response.data[0].error.message} during google_ads_offline_conversions response transformation`
+        )
+        .setAuthErrorCategory(
+          getAuthErrCategory(get(response, "response.response.status"))
+        )
+        .build();
+    }
+    const conversionActionId = get(
+      response,
+      "response.data.0.results.0.conversionAction.id"
+    );
+    if (!conversionActionId) {
+      throw new ErrorBuilder()
+        .setStatus(400)
+        .setMessage(
+          `[Google Ads Offline Conversions]:: Unable to find conversionActionId for conversion:${params.event}`
+        )
+        .build();
+    }
+    return conversionActionId;
+  });
+};
+
+// transformer proxy
+
+/**
+ * collect conversionActionId for conversionAction parameter
+ * @param {*} request
+ * @returns
+ */
+const ProxyRequest = async request => {
+  const { method, endpoint, headers, params, body } = request;
+
+  const conversionActionId = await getConversionActionId(
+    method,
+    headers,
+    params
+  );
```

Comment:
Why method is from proxyRequest is used here `getConversionAction` API would have a definite method which should be used